core/contracts/financial_templates/PricelessPositionManager.sol

```diff
@@ -197,11 +226,114 @@ contract PricelessPositionManager is Testable {
         tokenCurrency.burn(numTokens.rawValue);
     }
 
+    /**
+    * @notice After expiration of the contract the DVM is asked what for the prevailing price at the time of
+    * expiration.
+    */
+    function expire() public onlyPostExpiration() {
+        _requestOraclePrice(expirationTimestamp);
+    }
+
+    /**
+     * @notice After a contract has passed maturity all token holders can redeem their tokens for underlying at
+     * the prevailing price defined by the DVM from the `expire` function. This Burns all tokens from the caller
+     * of `tokenCurrency` and sends back the proportional amount of `collateralCurrency`.
+     */
+    function settleExpired() public onlyPostExpiration() {
+        // Get the current settlement price. If it is not resolved will revert.
+        FixedPoint.Unsigned memory settlementPrice = _getOraclePrice(expirationTimestamp);
+
+        // Get caller's tokens balance and calculate amount of underlying entitled to them.
+        FixedPoint.Unsigned memory tokensToRedeem = FixedPoint.Unsigned(tokenCurrency.balanceOf(msg.sender));
+        FixedPoint.Unsigned memory totalRedeemableCollateral = tokensToRedeem.div(settlementPrice);
+
+        // TODO: what happens in the case where the position is invalid but the contract has expired; i.e they were
+        // liquidated close to settlement and then contract settles before they withdraw.
+
+        // If the caller is a sponsor they are also entitled to their underlying excess collateral.
+        if (positions[msg.sender].isValid) {
+            PositionData storage positionData = _getPositionData(msg.sender);
+
+            // Calculate the underlying entitled to a token sponsor. This is collateral - debt in underlying.
+            FixedPoint.Unsigned memory tokenDebtValueInCollateral = positionData.tokensOutstanding.div(settlementPrice);
+            FixedPoint.Unsigned memory positionRedeemableCollateral = positionData.collateral.sub(
+                tokenDebtValueInCollateral
+            );
+
+            // Add the number of redeemable tokens for the sponsor to their total redeemable collateral.
+            totalRedeemableCollateral = totalRedeemableCollateral.add(positionRedeemableCollateral);
+
+            // Reset the position state as all the value has been removed after settlement.
+            positionData.collateral = FixedPoint.fromUnscaledUint(0);
```

Comment:
nit: instead of manually setting to `0`, can we just delete the positionData struct in its entirety?